### PR TITLE
New Resource - `azurerm_vmware_netapp_volume_attachment`

### PR DESCRIPTION
**Pull request by xiaxyi**
Adding new feature: Azure netapp file as VMware solution storage attachment.

local test:
```
--- PASS: TestAccVmwarePrivateCloudNetappFileVolumeAttachment_basic (14289.48s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/vmware        14290.140s
```

online testing is passing:
image

### CODE REVIEW

File: internal/services/vmware/vmware_private_cloud_netapp_file_volume_attachment_resource_test.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+func (r VmwareNetappFileVolumeAttachmentResource) templatePrivateCloud(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {
+    resource_group {
+      prevent_deletion_if_contains_resources = false
+    }
+  }
+  # In Vmware acctest, please disable correlation request id, else the continuous operations like update or delete will not be triggered
+  # issue https://github.com/Azure/azure-rest-api-specs/issues/14086
```
Comment (tombuildsstuff) [inline internal/services/vmware/vmware_private_cloud_netapp_file_volume_attachment_resource_test.go:78]: waiting on https://github.com/Azure/azure-rest-api-specs/issues/14086#issuecomment-1298625805

Comment (cataggar) [reply internal/services/vmware/vmware_private_cloud_netapp_file_volume_attachment_resource_test.go:78]: To be safe, I'd leave the workaround in place in November.

Comment (katbyte) [reply internal/services/vmware/vmware_private_cloud_netapp_file_volume_attachment_resource_test.go:78]: @cataggar - this is only for the tests and won't affect end users, we should be removing it now so the tests represent what a user experiences

**Comment by xiaxyi** [comment]
I removed the correlation_id and acc tests goes fine locally
--- PASS: TestAccVmwarePrivateCloudNetappFileVolumeAttachment_basics (15086.94s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/vmware        15087.678s

For teamCity testing, I already asked the service team to have 3 nodes provisioned for hc subscription, but I still need your help to get the [Prerequisites ](https://learn.microsoft.com/en-us/azure/azure-vmware/attach-azure-netapp-files-to-azure-vmware-solution-hosts?tabs=azure-portal#prerequisites)ready so I can trigger the online testing.
```
1. az feature register --name "ANFAvsDataStore" --namespace "Microsoft.NetApp"
2. register the avs feature:https://learn.microsoft.com/en-us/azure/azure-vmware/attach-azure-netapp-files-to-azure-vmware-solution-hosts?tabs=azure-portal#prerequisites
```

File: website/docs/r/vmware_private_cloud__netapp_file_attachment.html.markdown [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+---
+The following arguments are supported:
+
+* `name` - (Required) The name which should be used for this VMware Private Cloud Netapp File Volume Attachment. Changing this forces a new VMware Private Cloud Netapp File Volume Attachment to be created.
+
+* `netapp_volume_id` - (Required) The netapp file volume for this VMware Private Cloud Netapp File Volume Attachment to connect to. Changing this forces a new VMware Private Cloud Netapp File Volume Attachment to be created.
+
+* `vmware_cluster_id` - (Required) The vmware cluster for this VMware Private Cloud Netapp File Volume Attachment to associated to. Changing this forces a new VMware Private Cloud Netapp File Volume Attachment to be created.
+
+~> **NOTE :** please follow the prerequisites mentioned in this [article](https://learn.microsoft.com/en-us/azure/azure-vmware/attach-azure-netapp-files-to-azure-vmware-solution-hosts?tabs=azure-portal#prerequisites) before associating the netapp file volume to the vmware solution hosts.
```
Comment (katbyte) [inline website/docs/r/vmware_private_cloud__netapp_file_attachment.html.markdown:173]: this not should be up the top in the resource description before the example

**Comment by katbyte** [comment]
@xiaxyi - feature registered

**Comment by xiaxyi** [comment]
Thanks @katbyte , both online and local testing are passing, may I know if you have any other concerns regarding this pr?

**Comment by xiaxyi** [comment]
@paulomarquesc  I see the code in error (d.get in deletion func) was implemented by you in your pr #10610. Any idea about why to get the data protection from the user config, instead of the API directly?

**Review by katbyte — APPROVED** [review]
LGTM - we can fix the .get in delete another time

**Pull request merged**